src/lib.rs

```diff
@@ -474,3 +474,30 @@ impl Source for Error {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    struct Inner {
+        vec: Vec<u64>,
+        value: u64,
+    }
+
+    #[test]
+    fn test_strategy() {
```

Comment:
I think this unit test would be better as a doctest on `Strategy::wrap`. It would be a good way to document that the returned reference is guaranteed to point to the same location as the given pointer.